### ISSUE TITLE
hotbar appearing at top of screen

### ISSUE THREAD

**Issue by tenplus1** [minetest/minetest#3989]
After updating to build 4724 on linux the hotbar appears in the top left side of the screen instead of bottom-middle...

http://i63.tinypic.com/15wh0zt.jpg

**Comment by kwolekr** [comment]
What mods are you using?

**Comment by rubenwardy** [comment]
> I tried on both singeplayer with NO mods (looks like that picture) and on server with wuzzy's hudbars (looks same)
> so with or without mods it's not appearing in the right place

**Comment by Fixer-007** [comment]
Reproduced. Singleplayer, v6, start game, ???, top left corner hud.

**Comment by PilzAdam** [comment]
Offending commit is: eae33951ad8b13cb335fca8eaec73185959eb44e

**Comment by PilzAdam** [comment]
73d4538d9463b349eedc767c47d1750a164e2b7b

**Comment by Fixer-007** [comment]
Does not fix it for me, https://i.imgur.com/jy2VbUU.png
Please open.

**Comment by everamzah** [comment]
Entirely missing for me; perhaps related to DPI?

**Comment by everamzah** [comment]
With `xrandr --dpi 72` hotbar is middle-left.  Using `xrandr --dpi 96` results in the proper placement.  Some higher value, say 144 and the hotbar is entirely missing.

**Comment by rubenwardy** [comment]
From other code, it looks like it should be

```cpp
pos + offset * hud_scaling * density
```

rather than

```cpp
(pos + offset) * hud_scaling * density
```

This is probably a refactoring mistake by me.
Before the patch, hud scaling was not taken into account as there was no offset.

https://github.com/minetest/minetest/blob/73d4538d9463b349eedc767c47d1750a164e2b7b/src/hud.cpp#L385

**Comment by sofar** [comment]
commit 4fdf66bbf4d90846844ee3f6d41acb31ce06ca8b

![](http://i.imgur.com/yax8jR0.jpg)

This worked fine earlier.

**Comment by rubenwardy** [comment]
Found a fix. Preparing.

**Comment by rubenwardy** [comment]
PR merged, it should be fixed now? :/

**Comment by Fixer-007** [comment]
Will try it.

**Comment by sofar** [comment]
Confirmed fixed with b5f6d41eedca983a98470fdb97b5732e08802d24.
